src/client/voice/dispatcher/KryptonDispatcher.js

```diff
@@ -46,22 +47,23 @@ class KryptonDispatcher extends StreamDispatcher {
     super.pause();
   }
 
-  _write(chunk, enc, done) {
+  async _write(chunk, enc, done) {
     if (['opus', 'ogg/opus', 'webm/opus'].includes(this._type)) {
       super._write(chunk, enc, done);
     } else {
       this._buffer = Buffer.concat([this._buffer, chunk]);
 
-      const func = () => {
-        super._write(this._buffer.slice(0, 3840), enc, () => {
-          if (this._buffer.length >= 3840) func();
-          else done();
-        });
+      const write = promisify(super._write).bind(this);
```

Comment:
For performance, I think you can do

```javascript
const superWrite = promisify(StreamDispatcher.prototype._write);

// ... Inside KryptonDispatcher#_write
const write = superWrite.bind(this);
```

This way you would not need to promisify the function every time you call `KryptonDispatcher#_write`.

Reply:
I already fixed this.